net-p2p/retroshare/retroshare-0.6.9999.ebuild

```diff
@@ -14,131 +14,101 @@ LICENSE="GPL-2 GPL-3 Apache-2.0 LGPL-2.1"
+		dev-qt/qtscript:5
+		dev-qt/qtxml:5
 		dev-qt/qtgui:5
 		dev-qt/qtwidgets:5
 		dev-qt/qtx11extras:5
 		x11-libs/libX11
 		x11-libs/libXScrnSaver
 	)
-	voip? (
-		media-libs/opencv[-qt4(-)]
-		media-libs/speex
-		virtual/ffmpeg[encode]
-	)"
-DEPEND="${RDEPEND}
-	virtual/pkgconfig"
-
-src_prepare() {
-	local dir
-
-	sed -i \
-		-e "s|/usr/lib/retroshare/extensions6/|/usr/$(get_libdir)/${PN}/extensions6/|" \
-		libretroshare/src/rsserver/rsinit.cc \
-		|| die "sed on libretroshare/src/rsserver/rsinit.cc failed"
+	settings-api? ( dev-qt/qtcore:5 )
+	sqlcipher? ( dev-db/sqlcipher )
+	!sqlcipher? ( dev-db/sqlite )
+	webui? ( net-libs/libmicrohttpd )"
 
-	rs_src_dirs="libbitdht/src openpgpsdk/src libresapi/src libretroshare/src supportlibs/pegmarkdown"
-	use cli && rs_src_dirs="${rs_src_dirs} retroshare-nogui/src"
-	use feedreader && rs_src_dirs="${rs_src_dirs} plugins/FeedReader"
-	use gui && rs_src_dirs="${rs_src_dirs} retroshare-gui/src"
-	use voip && rs_src_dirs="${rs_src_dirs} plugins/VOIP"
-
-	# Force linking to sqlcipher ONLY
-	sed -i \
-		-e '/isEmpty(SQLCIPHER_OK) {/aerror(libsqlcipher not found)' \
-		retroshare-gui/src/retroshare-gui.pro \
-		retroshare-nogui/src/retroshare-nogui.pro || die 'sed on retroshare-gui/src/retroshare-gui.pro failed'
-
-	# Avoid openpgpsdk false dependency on qtgui
-	sed -i '2iQT -= gui' openpgpsdk/src/openpgpsdk.pro || die
-
-	eapply_user
-}
+DEPEND="${RDEPEND}
+	gui? ( dev-qt/designer:5 )
+	dev-qt/qtcore:5
+	virtual/pkgconfig
+"
 
 src_configure() {
-	for dir in ${rs_src_dirs} ; do
-		pushd "${S}/${dir}" >/dev/null || die
-		eqmake5 $(use gnome-keyring && echo CONFIG+=rs_autologin)
-		popd >/dev/null || die
-	done
-}
+	local qConfigs=()
 
-src_compile() {
-	local dir
-
-	for dir in ${rs_src_dirs} ; do
-		emake -C "${dir}"
-	done
+	use cli && qConfigs+="retroshare_nogui" || qConfigs+="no_retroshare_nogui"
+	use control-socket && qConfigs+="libresapilocalserver"
+	use gnome-keyring && qConfigs+="rs_autologin"
+	use gui && qConfigs+="retroshare_gui" || qConfigs+="no_retroshare_gui"
+	use settings-api && qConfigs+="libresapi_settings"
+	use sqlcipher && qConfigs+="sqlcipher" || qConfigs+="no_sqlcipher"
+	use webui && qConfigs+="libresapihttpserver" || qConfigs+="no_libresapihttpserver"
 
-	unset rs_src_dirs
+	eqmake5 CONFIG+="${qConfigs[@]}"
 }
 
 src_install() {
-	local i
-	local extension_dir="/usr/$(get_libdir)/${PN}/extensions6/"
-
 	use cli && dobin retroshare-nogui/src/retroshare-nogui
 	use gui && dobin retroshare-gui/src/retroshare
 
-	exeinto "${extension_dir}"
-	use feedreader && doexe plugins/FeedReader/*.so*
-	use voip && doexe plugins/VOIP/*.so*
-
 	insinto /usr/share/retroshare
 	doins libbitdht/src/bitdht/bdboot.txt
 
-	doins -r libresapi/src/webui
+	use webui && doins -r libresapi/src/webui
 
 	dodoc README.md
 	make_desktop_entry retroshare
+
 	for i in 24 48 64 128 ; do
 		doicon -s ${i} "data/${i}x${i}/apps/retroshare.png"
 	done
 }
 
+pkg_pretend() {
+	if ! use sqlcipher; then
```

Comment:
Not convinced about it. Do you really think it is relevant when building a binary package that will be installed on other systems?

Reply:
So maybe we should warn at installation time too?

Reply:
`pkg_pretend` is called both when building binary packages and when installing.

Reply:
That seems the best from a RetroShare point of view ;)